helix-term/src/ui/mod.rs

```diff
@@ -83,7 +83,11 @@ pub fn regex_prompt(
 
                             fun(view, doc, regex, event);
 
-                            view.ensure_cursor_in_view(doc, cx.editor.config.scrolloff);
+                            let cursor = doc
+                                .selection(view.id)
+                                .primary()
+                                .cursor(doc.text().slice(..));
+                            view.ensure_cursor_in_view(cursor, doc, cx.editor.config.scrolloff);
```

Comment:
Why this change? You're calling it everywhere with the same primary cursor. Just fetch it inside the `ensure_cursor_in_view` method

Reply:
I had to use the use the ensuring cursor in view functionality for locations other than that of the primary cursor in the jump mode functionality (when searching for jump locations, it has to stop once it reaches a location outside the screen). I suppose in order to avoid the resulting duplication in other locations that use this functionality, we could add a shortcut method, so you could do `doc.primary_cursor(view.id)`?

Reply:
`is_cursor_in_view` seems to be used with other cursors, but I can't spot any ensure_ call that doesn't use the primary cursor

Reply:
Ah